### PR TITLE
Solution-03-03-2024

### PR DESCRIPTION
**Pull request by shafran-andriy**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+            "name": error.get("code"), "source": "flake8"
+            }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "path": file_path,
+        "errors": [format_linter_error(i) for i in errors],
```
Comment (vasylhnatiuk) [inline app/main.py]: i is a bad name

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+        "status": "failed" if any(errors) else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {"errors": [],
+         "path": "./test_source_code_2.py",
```
Comment (vasylhnatiuk) [inline app/main.py]: why do you use hard code? you can get path from `linter_report`

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
add venvScriptactivate in .gitignore and try delete it from pull request

File: app/main.py (hunk 1 above)
Comment (shafran-andriy) [reply app/main.py]: fixed: name i changed to error

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
fix tests

File: app/main.py (hunk 1 above)
Comment (shafran-andriy) [reply app/main.py]: venvScriptactivate added in .gitignore

File: app/main.py (hunk 2 above)
Comment (shafran-andriy) [reply app/main.py]: "path": linter_report.get("./test_source_code_2.py") - changed 
But I do not know if it right

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
fix tests. if you have questions write them in chat

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
+    return [
+        {"errors": [],
+         "path": linter_report.get("./test_source_code_2.py"),
+         "status": "passed"
+         },
+        format_single_linter_file(
+            "./source_code_2.py",
+            linter_report.get("./source_code_2.py")
+        )
+    ]
```
Comment (EdAlekseiev) [inline app/main.py]: You should iterate by linter_report values
`
return [
        format_single_linter_file(path, errors)
        for (path, errors) in linter_report.items()
    ]
`

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 3 above)
Comment (shafran-andriy) [reply app/main.py]: fixed

**Review by EdAlekseiev — APPROVED** [review]
Good job

**Pull request closed without merging**